lib/xcpretty/reporters/json_compilation_database.rb

```diff
@@ -21,36 +21,44 @@ def load_dependencies
 
     def initialize(options)
       load_dependencies
-      @filepath           = options[:path] || FILEPATH
-      @parser             = Parser.new(self)
-      @compilation_units  = []
-      @current_file       = nil
-      @current_path       = nil
+      @file_path = options[:path] || FILE_PATH
+      @parser = Parser.new(self)
+      @compilation_units = []
+      @pch_path = nil
+      @current_file = nil
+      @current_path = nil
     end
 
     def handle(line)
       @parser.parse(line)
     end
 
+    def format_process_pch_command(file_path)
+      @pch_path = file_path
+    end
+
     def format_compile(file_name, file_path)
       @current_file = file_name
-      @current_path = File.dirname(file_path)
+      @current_path = file_path
     end
 
-    def format_compile_command(compiler_command)
-      @compilation_units << { :command => compiler_command, 
-        :file => @current_file, :directory => @current_path }
+    def format_compile_command(compiler_command, file_path)
+      directory = file_path.gsub("#{@current_path}", '').gsub(/\/$/, '')
+      directory = '/' if directory.empty?
+      @compilation_units << {:command => compiler_command.gsub(/(\-include)\s.*\.pch/, "\\1 #{@pch_path}"),
```

Comment:
Line is too long. [107/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.